django/db/models/fields/__init__.py

```diff
@@ -237,14 +237,15 @@ def _check_field_name(self):
         else:
             return []
 
+    @classmethod
+    def is_value(cls, value, accept_promise=True):
+        return isinstance(value, (str, Promise) if accept_promise else str) or not is_iterable(value)
+
     def _check_choices(self):
         if not self.choices:
             return []
 
-        def is_value(value, accept_promise=True):
-            return isinstance(value, (str, Promise) if accept_promise else str) or not is_iterable(value)
-
-        if is_value(self.choices, accept_promise=False):
```

Comment:
We use `accept_promise=False` only here it's misleading to me, I opened #12000 to fix this.